File: Library/PaintedSkin/src/main/java/org/alee/component/skin/exception/ApplySkinException.java
package org.alee.component.skin.exception;

import android.view.View;

import org.alee.component.skin.executor.SkinElement;

/**********************************************************
 *
 * @author: MY.Liu
 * @date: 2021/2/13
 * @description: xxxx
 *
 *********************************************************/
public final class ApplySkinException extends RuntimeException {

    public ApplySkinException(View view, SkinElement element, Throwable throwable) {
        super("An exception occurred at apply theme skin ! *********" +
                " \r\n【 ViewId: " + view.getId() + " 】 【 ViewName: " + view.getClass().getName() + " 】" +
                "\r\n 【 AttributeName: " + element.getAttrName() + " 】 【 ResourceId: " + element.getResourcesId() + " 】 " +
                "【 ResourceName: " + element.getResourcesName() + " 】 【ResourceType: " + element.getResourcesType() + " 】", throwable);
    }
}

File: Library/PaintedSkin-Kotlin/src/main/java/org/alee/component/skin/exception/ApplySkinException.kt
package org.alee.component.skin.exception

import android.view.View
import org.alee.component.skin.model.SkinAttribute
import org.alee.component.skin.util.INotProguard

/**
 * 换肤失败异常
 *
 * <p> 详细描述
 *
 * @author MingYu.Liu
 * created in 2023/9/1
 *
 */
class ApplySkinException(view: View, attribute: SkinAttribute, throwable: Throwable) :
    RuntimeException(
        """An exception occurred at apply theme skin ! *********
        [ViewId: ${view.id}] [ViewName: ${view.javaClass.name}]
        [AttributeName: ${attribute.attrName}] [ResourceId: ${attribute.resourceId}]
        [ResourceName: ${attribute.resourceName}] [ResourceType: ${attribute.resourceType}]
    """,
        throwable,
    ),
    INotProguard
